src/misc/lv_style.h

```diff
@@ -152,10 +168,38 @@ typedef struct {
 
 /** A descriptor of a gradient. */
 typedef struct {
-    lv_gradient_stop_t   stops[LV_GRADIENT_MAX_STOPS]; /**< A gradient stop array */
-    uint8_t              stops_count;                  /**< The number of used stops in the array */
-    lv_grad_dir_t        dir : 3;                      /**< The gradient direction.
-                                                        * Any of LV_GRAD_DIR_HOR, LV_GRAD_DIR_VER, LV_GRAD_DIR_NONE */
+    lv_gradient_stop_t   stops[LV_GRADIENT_MAX_STOPS];  /**< A gradient stop array */
+    uint8_t              stops_count;                   /**< The number of used stops in the array */
+    lv_grad_dir_t        dir : 3;                       /**< The gradient direction.
+                                                         * Any of LV_GRAD_DIR_NONE, LV_GRAD_DIR_VER, LV_GRAD_DIR_HOR,
+                                                           LV_GRAD_TYPE_LINEAR, LV_GRAD_TYPE_RADIAL, LV_GRAD_TYPE_CONICAL */
+    lv_grad_extend_t     extend : 2;                    /**< Behaviour outside the defined range.
+                                                         * LV_GRAD_EXTEND_NONE, LV_GRAD_EXTEND_PAD, LV_GRAD_EXTEND_REPEAT, LV_GRAD_EXTEND_REFLECT */
+#if LV_USE_DRAW_SW_COMPLEX_GRADIENTS
+    union {
+        /*Linear gradient parameters*/
+        struct {
+            lv_point_t  start;                          /**< Linear gradient vector start point */
+            lv_point_t  end;                            /**< Linear gradient vector end point */
+        } linear;
+        /*Radial gradient parameters*/
+        struct {
+            lv_point_t  focal;                          /**< Center of the focal (starting) circle in local coordinates */
+            /* (can be the same as the ending circle to create concentric circles) */
+            lv_point_t  focal_extent;                   /**< Point on the circle (can be the same as the center) */
+            lv_point_t  end;                            /**< Center of the ending circle in local coordinates */
+            lv_point_t  end_extent;                     /**< Point on the circle determining the radius of the gradient */
+        } radial;
+        /*Conical gradient parameters*/
+        struct {
+            lv_point_t  center;                         /**< Conical gradient center point */
+            int16_t     start_angle;                    /**< Start angle 0..3600 */
+            int16_t     end_angle;                      /**< End angle 0..3600 */
+        } conical;
+    } params;
+    void * state;
+#endif
+    struct _lv_grad_dsc_t * next;                       /**< Next gradient to render more gradients*/
```

Comment:
Adding this linked list pointer was originally suggested by Gabor. It is not yet implemented, but it would be a useful feature. Maybe we can remove it for now.